cowprotocol/cowswap
# [ETH-FLOW-V2] Deadline enforcement COMPONENTS

# Summary

Part of / merges into #1250 

Adds components to display enforcements from #1250

### MINIMUM THRESHOLD IS CURRENTLY A PLACEHOLDER @josojo AT 10 MINUTES (this will change probably)

![image](https://user-images.githubusercontent.com/21335563/197833831-3bd86d23-a54f-4c8f-a235-77fc152753f8.png)


## QA / Process:
user has deadline set at 6 minutes
user is ERC20 <> ERC20
all good, user can swap fine with no UX/UI change as nothing is enforced
user switches to NATIVE <> ERC20 (sell native)
user sees Transaction deadline (modified) 10 minutes
user opens settings, tries to set deadline back to 6 minutes user cannot
tooltip updated to relevant info
user sets deadline to x minutes in settings (X> deadline)
user should not see any warning or mentioning of the deadline
